Add traffic requirements documentation (#8349)
istio/istio.io: content/en/docs/ops/deployment/requirements/index.md

| 15014 | HTTP | Istiod | Control plane monitoring |

To avoid port conflicts with sidecars, applications should not use any of the ports used by Envoy.

## Server First Protocols

Some protocols are "Server First" protocols, which means the server will send the first bytes. This may have an impact on
[`PERMISSIVE`](/docs/reference/config/security/peer_authentication/#PeerAuthentication-MutualTLS-Mode) mTLS and [Automatic protocol selection](/docs/ops/configuration/traffic-management/protocol-selection/#automatic-protocol-selection).

Both of these features work by inspecting the initial bytes of a connection to determine the protocol, which is incompatible with server first protocols.

In order to support these cases, follow the [Explicit protocol selection](/docs/ops/configuration/traffic-management/protocol-selection/#explicit-protocol-selection) steps to declare the protocol of the application as `TCP`.

> The following ports are known to be commonly carry server first protocols, and are automatically assumed to be `TCP`:

Reviewer: ```suggestion
The following ports are known to commonly carry server first protocols, and are automatically assumed to be `TCP`:
```

|Protocol|Port|
|--------|----|
| SMTP   |25  |
| DNS    |53  |
| MySQL  |3306|
| MongoDB|27017|

> Additionally, TLS configuration must be configured. Because TLS communication is not server first, TLS encrypted server first traffic will not have issues with protocol detection. There are a few options to get things working

Reviewer: I'm still not sure I completely understand what you're trying to say. Let me know if this is right.
```suggestion
Because TLS communication is not server first, TLS encrypted server first traffic will work with automatic protocol detection as long as you make sure that all traffic subjected to TLS sniffing is encrypted:
```

> 1. Configure `mTLS` mode `STRICT` for the server. This will enforce all traffic is TLS encrypted.

Reviewer: ```suggestion
1. Configure `mTLS` mode `STRICT` for the server. This will ensure all traffic is TLS encrypted.
```

Author: This does not ensure, clients can send plaintext and it will be blocked. This is why I went with "enforce"

Reviewer: I was just trying to improve the grammar, unfortunately with a small semantic change :(
How about:
```suggestion
1. Configure `mTLS` mode `STRICT` for the server. This will enforce TLS encryption for all requests.
```

> 1. Configure `mTLS` mode `DISABLE` for the server. This will disable the TLS sniffing, allowing server first protocols to be used

Reviewer: ```suggestion
1. Configure `mTLS` mode `DISABLE` for the server. This will disable the TLS sniffing, allowing server first protocols to be used.
```

> 1. Configure all clients to send `TLS` traffic, generally through a [`DestinationRule`](/docs/reference/config/networking/destination-rule/#ClientTLSSettings) or relying on Auto mTLS.

Reviewer: ```suggestion
1. Configure all clients to send `TLS` traffic, generally through a [`DestinationRule`](/docs/reference/config/networking/destination-rule/#ClientTLSSettings) or by relying on auto mTLS.
```
